src/packagedcode/pypi.py

```diff
@@ -303,6 +305,45 @@ def parse_setup_py(location):
     )
 
 
+def parse_requirements_txt(location):
+    """
+    Return a Package built from requirements.txt.
+    """
+    package_dependencies = []
+    scope = 'dependencies'
+    runtime = True
+    optional = False
+    if 'test.txt' in location:
+        scope = 'test'
+        runtime = False
+        optional = True
+    elif 'dev.txt' in location:
+        scope = 'dev'
+        runtime = False
+        optional = True
+    with io.open(location, encoding='utf-8') as loc:
+        for req in requirements.parse(loc):
+            resolved = False
+            if '==' in (req.specs)[0]:
```

Comment:
I think `(req.specs)[0]` can be made into a variable since you reference it in multiple places after this, like line 334 and line 336.

Also, does this work if a requirement.txt requirement does not state a version? For example, consider the following as a requirements.txt:

```
setuptools==32.0.0
django
```

Would django be processed properly?

Reply:
I agree with @JonoYang. I think your approach is slightly flawed here.

Reply:
@JonoYang You were right. I have made some changes :)